src/net/sourceforge/kolmafia/session/ChoiceAdventures.java

```diff
@@ -9079,16 +9079,39 @@ else if (KoLCharacter.isEd()) {
     return null;
   }
 
+  private static Map<String, String> defenseToElement =
+      Map.ofEntries(
+          Map.entry("firewall", "hot"),
+          Map.entry("ICE barrier", "cold"),
+          Map.entry("corruption quarantine", "stench"),
+          Map.entry("parental controls", "sleaze"),
+          Map.entry("null container", "spooky"));
+
+  // This is incomplete
+  private static Map<String, String> encounterToElement =
+      Map.ofEntries(
+          Map.entry("Ice Cream Antisocial", "cold"),
```

Comment:
There are more on the wiki at https://kol.coldfront.net/thekolwiki/index.php/Cyber_-_Halfway

Reply:
Sweet. I updated the source and tests to have everything they listed.

That list is still almost certainly incomplete: there are 2 hot, 4 cold, 2 stench, 2 sleaze, and 4 spooky.
But, we can always add more later.